README.md

```diff
@@ -162,6 +162,14 @@ $.Date :: Type
 
 Type comprising every Date value.
 
+#### `Enum`
+
+```haskell
+$.Enum :: [Any] -> Type
+```
+
+Constructor for [enumerated types][2].
+
```

Comment:
Let's move this to the [type constructors][1] section—just before [`$.RecordType`][2], I suggest—and expand the documentation to match the format used for the other type constructors.


[1]: https://github.com/plaid/sanctuary-def/blob/master/README.md#type-constructors
[2]: https://github.com/plaid/sanctuary-def/blob/master/README.md#recordtype